examples/kitty/client/kitty.py

```diff
@@ -253,21 +265,44 @@ async def read_from_server():
             continue
 
         info(message)
-        words = message.decode('utf-8').split()
+
+        words = message.decode('utf-8').split(" ", 1)
         if words[0] == '101':
             token = int(words[1])
             info(f'TOKEN = {token}')
-        elif words[0] == '200':
-            card_id = words[2]
-            display.rect(200, 400, 1000, 1000, 0x0, True)
-            wri.set_textpos(display, 400, 260)
-            wri.printstring("Thanks for your purchase!")
-            wri.set_textpos(display, 470, 235)
-            wri.printstring(f"Card UID: {card_id}")
+        
+        # Print server time in smaller font
+        elif words[0] == '100':
+            display.rect(500, 250, 1000, 100, 0x0, True)
+            wri35.set_textpos(display, 250, 500)
+            wri35.printstring(words[1])
             display.show()
-            asyncio.create_task(wait_seconds_and_call(3, reset_status))
 
 
+        else:
+            # Print server response
+            display.rect(0, 400, 2000, 2000, 0x0, True)
+            # Position text in middle of row
+            wri50.set_textpos(display, 430, (TRUE_DISPLAY_WIDTH - wri50.stringlen(words[1]))//2)
+            wri50.printstring(f"{words[1]}")
+            display.show()
+
+            # Display "new card" message for longer, so person has enough time to see number
+            if words[0] == '400':
+                message_display_time = 5
+            else:
+                message_display_time = 3
+
+            # Printing "-$1" if there has been a purchase
+            if words[0] == '200':
+                display.rect(0, 500, 2000, 2000, 0x0, True)
+                wri50.set_textpos(display, 500, (TRUE_DISPLAY_WIDTH - wri50.stringlen(f'-$1'))//2)
+                
+                wri50.printstring(f"-$1")
+                display.show()
```

Comment:
Can we have one `display.show()` at the end of this conditional to make sure the framebuffer is updated once.